vnext/Microsoft.ReactNative/ReactHost/ReactInstanceWin.cpp

```diff
@@ -539,6 +543,12 @@ Mso::Future<void> ReactInstanceWin::Destroy() noexcept {
 
   // Make sure that the instance is not destroyed yet
   if (auto instance = m_instance.Exchange(nullptr)) {
+    {
+      // Release the JSI runtime
+      std::scoped_lock lock{m_mutex};
```

Comment:
Nit: Should we name the mutex `m_jsiRuntimeMutex` to clarify?

Reply:
No, it is not JSI Runtime specific. This class has only a single mutex for all fields to avoid any possible deadlocks.